net_transport.go

```diff
@@ -163,11 +189,18 @@ func NewNetworkTransportWithConfig(
 			Level:  hclog.DefaultLevel,
 		})
 	}
+	maxInFlight := config.MaxRPCsInFlight
+	if maxInFlight < 1 {
+		// Default (zero or nonsense negative value given) to 2 (which translates to
+		// a zero length channel buffer)
+		maxInFlight = 2
```

Comment:
🤔 I was just doing this, and I think we should make this case a Constant like the other defaults.

But I realised that all the `2`s in newNetPipeline and `AppendEntriesPipeline` probably should _not_ be changed. The rationale is: they are are 2 because 2 is the minimum value that is possible in our current implementation - they are guarding against panics or using code that can't work with a value specifically less than 2.

For example if we ever decided that the best default was actually 1 (disabling pipelining) or maybe a bit more than two,  every usage except this one should still be 2 since that's a property of the actual code implementation not just the default value.

Do you think it's work making two separate constants something like `DefaultMaxInFlight = 2` and `minValidMaxInFlight = 2` so that we can show that we are using the 2 to protect the implementation for doing something invalid in those other cases not necessarily because it's the default?

Reply:
I just pushed a commit with the approach above - two separate constants to express the semantics of the subtly-different `2` values!